File: 735-AsteroidCollision/735-AsteroidCollision.java
// Last updated: 8/13/2025, 8:19:16 PM
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
    Stack<Integer> stack = new Stack<>();
    int n = asteroids.length;
    int i=0;
    while(i < n){
        if(asteroids[i]<0){
            if(stack.size()==0 || stack.peek()<0){
                stack.push(asteroids[i]);
                i++;
            }
            else if(stack.peek()>Math.abs(asteroids[i])){
                i++;
            }
            else if(stack.peek()<Math.abs(asteroids[i])){
                stack.pop();
            }
            else{
                stack.pop();
                i++;
            }
        }
        else{
            stack.push(asteroids[i]);
            i++;
        }
}
int[] ans = new int[stack.size()];
for(int j=ans.length-1;j>=0;j--){
ans[j] = stack.pop();


}
return ans;
}

}

File: 735-AsteroidCollision/735-AsteroidCollision.py
# Last updated: 8/13/2025, 8:19:10 PM
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        stack = []

        for a in asteroids : 
            while stack and stack[-1] > 0 and a < 0 and stack[-1] < abs(a):
                stack.pop()

            if not stack or stack[-1] < 0 or a > 0:
                stack.append(a)
            elif stack[-1] == -a:
                stack.pop()  # same size, both destroyed

        
        return stack